pkg/machines/components/networks/createNetworkDialog.jsx

```diff
@@ -0,0 +1,560 @@
+
+/*
+ * This file is part of Cockpit.
+ *
+ * Copyright (C) 2019 Red Hat, Inc.
+ *
+ * Cockpit is free software; you can redistribute it and/or modify it
+ * under the terms of the GNU Lesser General Public License as published by
+ * the Free Software Foundation; either version 2.1 of the License, or
+ * (at your option) any later version.
+ *
+ * Cockpit is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details.
+ *
+ * You should have received a copy of the GNU Lesser General Public License
+ * along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React from 'react';
+import PropTypes from 'prop-types';
+import { Button, FormGroup, HelpBlock, Modal } from 'patternfly-react';
+
+import { MachinesConnectionSelector } from '../machinesConnectionSelector.jsx';
+import { ModalError } from 'cockpit-components-inline-notification.jsx';
+import { networkCreate } from '../../libvirt-dbus.js';
+import * as Select from 'cockpit-components-select.jsx';
+import { isEmpty, LIBVIRT_SYSTEM_CONNECTION, rephraseUI } from '../../helpers.js';
+import * as utils from './utils';
+import cockpit from 'cockpit';
+
+import './createNetworkDialog.css';
+
+const _ = cockpit.gettext;
+
+function validateParams(dialogValues) {
+    let validationFailed = {};
+
+    if (isEmpty(dialogValues.name.trim())) {
+        validationFailed['name'] = _("Name should not be empty");
+    }
+
+    if (dialogValues.ip === "IPv4 only" || dialogValues.ip === "IPv4 and IPv6") {
+        if (isEmpty(dialogValues.ipv4.trim())) {
+            validationFailed['ipv4'] = _("IPv4 Network should not be empty");
+        } else if (!utils.validateIpv4(dialogValues.ipv4)) {
+            validationFailed['ipv4'] = _("Invalid IPv4 address");
+        }
+
+        if (isEmpty(dialogValues.netmask.trim())) {
+            validationFailed['netmask'] = _("Mask or Prefix Length should not be empty");
+        } else if (!utils.validateNetmask(dialogValues.netmask)) {
+            validationFailed['netmask'] = _("Invalid IPv4 mask or prefix length");
+        }
+
+        if (dialogValues.ipv4DhcpEnabled) {
+            if (isEmpty(dialogValues.ipv4DhcpRangeStart.trim())) {
+                validationFailed['ipv4DhcpRangeStart'] = _("Start should not be empty");
+            } else if (!utils.validateIpv4(dialogValues.ipv4DhcpRangeStart)) {
+                validationFailed['ipv4DhcpRangeStart'] = _("Invalid IPv4 address");
+            } else if (!utils.isIpv4InNetwork(dialogValues.ipv4, dialogValues.netmask, dialogValues.ipv4DhcpRangeStart)) {
+                validationFailed['ipv4DhcpRangeStart'] = _("Address not within subnet");
+            }
+
+            if (isEmpty(dialogValues.ipv4DhcpRangeEnd.trim())) {
```

Comment:
In general we don't use curly brackets for one liners.
If even one body of the condition clause is though multiline we use brackets.
I don't really care tbh, so you can leave it as is.
It's just a common convention.